lib/ecto/query.ex

```diff
@@ -337,6 +337,9 @@ defmodule Ecto.Query do
 
   defmodule DynamicExpr do
     @moduledoc false
+
+    @type t :: %__MODULE__{}
```

Comment:
Since this is private, I don't think we should expose a type. What if we add `Ecto.Query.dynamic` instead? It should also be opaque, so something like:

    @opaque dynamic :: %DynamicExpr{}

Reply:
Sounds good :)